I'm trying to verify if cereal/psqueues are still a problem but I'm getting the build failure below. I found a similar problem here: https://github.com/NixOS/nixpkgs/pull/50920#issuecomment-441478643

```
building [html]: targets for 4 source files that are out of date
updating environment: 4 added, 0 changed, 0 removed
reading sources... [ 25%] index
reading sources... [ 50%] intro
reading sources... [ 75%] invoking

Extension error:
The 'ghc-flag' directive is already registered to domain std
reading sources... [100%] markup
clang-5.0: warning: argument unused during compilation: '-nopie' [-Wunused-command-line-argument]
make[1]: *** [docs/users_guide/ghc.mk:16: docs/users_guide/build-html/users_guide/index.html] Error 2
make[1]: *** Waiting for unfinished jobs....
clang-5.0: warning: argument unused during compilation: '-nopie' [-Wunused-command-line-argument]

Extension error:
The 'ghc-flag' directive is already registered to domain std

looking for now-outdated files... none found
pickling environment... done
checking consistency... done
preparing documents... done
writing output... [ 25%] index
make[1]: *** [docs/users_guide/ghc.mk:28: docs/users_guide/build-man/ghc.1] Error 2
writing output... [ 50%] intro
writing output... [ 75%] invoking
clang-5.0: warning: argument unused during compilation: '-nopie' [-Wunused-command-line-argument]
<<ghc: 20003168 bytes, 11 GCs, 233684/298496 avg/max bytes residency (2 samples), 24M in use, 0.000 INIT (0.004 elapsed), 0.021 MUT (0.876 elapsed), 0.019 GC (0.028 elapsed) :ghc>>
writing output... [100%] markup

/private/tmp/501/nix-build-ghc-8.4.4.drv-0/ghc-8.4.4/utils/haddock/doc/invoking.rst:457: WARNING: unknown option: -cpp
generating indices... genindex
writing additional pages... search
copying static files... done
copying extra files... done
dumping search index in English (code: en) ... done
dumping object inventory... done
build succeeded, 1 warning.

The HTML pages are in .build-html.
cp -R utils/haddock/doc/.build-html utils/haddock/doc/haddock
make: *** [Makefile:127: all] Error 2
builder for '/nix/store/h7zflbah2x7n9zqcpfif3x3wzb12hqj2-ghc-8.4.4.drv' failed with exit code 2
cannot build derivation '/nix/store/fd4jy5xjyjaj02wv3y8g8bwvabbhd95k-purescript-0.12.2.drv': 1 dependencies couldn't be built

```

--- a/pkgs/development/haskell-modules/configuration-ghc-8.4.x.nix
+++ b/pkgs/development/haskell-modules/configuration-ghc-8.4.x.nix
@@ -39,6 +39,10 @@ self: super: {
   unix = null;
   xhtml = null;
 
+  # Restricts aeson to <1.4
+  # https://github.com/purescript/purescript/pull/3537
+  purescript = doJailbreak super.purescript;
+
   # https://github.com/jcristovao/enclosed-exceptions/issues/12
   enclosed-exceptions = dontCheck super.enclosed-exceptions;
 

--- a/pkgs/top-level/all-packages.nix
+++ b/pkgs/top-level/all-packages.nix
@@ -7045,7 +7045,8 @@ in
 
   all-cabal-hashes = callPackage ../data/misc/hackage { };
 
-  purescript = haskell.lib.justStaticExecutables haskellPackages.purescript;
+  # Build with ghc 8.4 due to https://github.com/NixOS/nixpkgs/issues/53597
+  purescript = haskell.lib.justStaticExecutables haskell.packages.ghc844.purescript;
   psc-package = haskell.lib.justStaticExecutables
     (haskellPackages.callPackage ../development/compilers/purescript/psc-package { });
 

--- a/pkgs/top-level/haskell-packages.nix
+++ b/pkgs/top-level/haskell-packages.nix
@@ -49,13 +49,13 @@ in {
 
     ghc822 = callPackage ../development/compilers/ghc/8.2.2.nix {
       bootPkgs = packages.ghc822Binary;
-      inherit (buildPackages.python3Packages) sphinx;
+      sphinx = buildPackages.python3Packages.sphinx_1_7_9;
       buildLlvmPackages = buildPackages.llvmPackages_39;
       llvmPackages = pkgs.llvmPackages_39;
     };
     ghc844 = callPackage ../development/compilers/ghc/8.4.4.nix {
       bootPkgs = packages.ghc822Binary;
-      sphinx = null;            # broken with sphinx 1.8.x
+      sphinx = buildPackages.python3Packages.sphinx_1_7_9;
       buildLlvmPackages = buildPackages.llvmPackages_5;
       llvmPackages = pkgs.llvmPackages_5;
     };

--- a/pkgs/top-level/python-packages.nix
+++ b/pkgs/top-level/python-packages.nix
@@ -4116,6 +4116,17 @@ in {
 
   sphinx = callPackage ../development/python-modules/sphinx { };
 
+  # sphinx update to 1.8.x broke ghc versions < 8.6
+  sphinx_1_7_9 = self.sphinx.overridePythonAttrs
+    {
+      version = "1.7.9";
+      src = fetchPypi {
+        pname = "Sphinx";
+        inherit version;
+        sha256 = "217a7705adcb573da5bbe1e0f5cab4fa0bd89fd9342c9159121746f593c2d5a4";
+      };
+    };
+
   sphinx-argparse = callPackage ../development/python-modules/sphinx-argparse { };
 
   sphinxcontrib-websupport = callPackage ../development/python-modules/sphinxcontrib-websupport { };

